src/pages/PrivateNotes/PrivateNotesListPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {useMemo} from 'react';
+import {ScrollView} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import withReportAndPrivateNotesOrNotFound from '@pages/home/report/withReportAndPrivateNotesOrNotFound';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {PersonalDetails, Report, Session} from '@src/types/onyx';
+
+type PrivateNotesListPageOnyxProps = {
+    /* Onyx Props */
+
+    /** All of the personal details for everyone */
+    personalDetailsList: OnyxCollection<PersonalDetails>;
+
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+type PrivateNotesListPageProps = PrivateNotesListPageOnyxProps & {
+    /** The report currently being looked at */
+    report: Report;
+};
+
+type NoteListItem = {
+    title: string;
+    action: () => void;
+    brickRoadIndicator: ValueOf<typeof CONST.BRICK_ROAD_INDICATOR_STATUS> | undefined;
+    note: string;
+    disabled: boolean;
+};
+
+function PrivateNotesListPage({report, personalDetailsList, session}: PrivateNotesListPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    /**
+     * Gets the menu item for each workspace
+     */
+    function getMenuItem(item: NoteListItem) {
+        return (
+            <MenuItemWithTopDescription
+                key={item.title}
+                description={item.title}
+                title={item.note}
+                onPress={item.action}
+                shouldShowRightIcon={!item.disabled}
+                numberOfLinesTitle={0}
+                shouldRenderAsHTML
+                brickRoadIndicator={item.brickRoadIndicator}
+                disabled={item.disabled}
+                shouldGreyOutWhenDisabled={false}
+            />
+        );
+    }
+
+    /**
+     * Returns a list of private notes on the given chat report
+     */
+    const privateNotes = useMemo(() => {
+        const privateNoteBrickRoadIndicator = (accountID: number) => (report.privateNotes?.[accountID].errors ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined);
+        return Object.keys(report.privateNotes ?? {}).map((accountID: string) => {
+            const privateNote = report.privateNotes?.[Number(accountID)];
```

Comment:
```suggestion
            const privateNote = report.privateNotes?.[accountID];
```
Do we need to convert this to number, to access the value?

Reply:
Yes, `privateNotes` type in the report is `Record<number, Note>` and it expects to be a number.